Mathlib/Topology/Algebra/Category/ProfiniteGrp.lean

```diff
@@ -161,4 +165,169 @@ instance : HasForget₂ ProfiniteGrp Grp where
     map := fun f => f.toMonoidHom
   }
 
+/-- A topological group that has a ContinuousMulEquiv to a profinite group is profinite. -/
+def ofContinuousMulEquivProfiniteGrp {G : ProfiniteGrp.{u}} {H : Type v} [TopologicalSpace H]
+    [Group H] [TopologicalGroup H] (e : ContinuousMulEquiv G H) : ProfiniteGrp.{v} :=
+  letI : CompactSpace H := Homeomorph.compactSpace e.toHomeomorph
+  letI : TotallyDisconnectedSpace G := Profinite.instTotallyDisconnectedSpaceαTopologicalSpaceToTop
+  letI : TotallyDisconnectedSpace H := Homeomorph.totallyDisconnectedSpace e.toHomeomorph
+  .of H
+
+/-- A closed subgroup of a profinite group is profinite. -/
+def ofClosedSubgroup {G : ProfiniteGrp}
+    (H : Subgroup G) (hH : IsClosed (H : Set G)) : ProfiniteGrp :=
+  letI : CompactSpace H := isCompact_iff_compactSpace.mp (IsClosed.isCompact hH)
+  of H
+
+end ProfiniteGrp
+
+/-!
+# The projective limit of finite groups is profinite
+
+* `FiniteGrp.limit` : the concretely constructed limit of finite group as a subgroup of the pi-type
+
+* `ofFiniteGrpLimit`: direct limit of finite groups is a profinite group
```

Comment:
Do you mean inverse limit? At any rate, you should use a consistent wording: either injective/projective or direct/inverse (possibly agreeing with what is already used in other parts of the library).

Reply:
Sorry that I don't know the math clearly. I think I prove it satisfies the universal property.